HW1/P4/P4.py

```diff
@@ -0,0 +1,29 @@
+from pyspark import SparkContext
+
+import networkx as nx
+import matplotlib.pyplot as plt
+import itertools
+import pdb
+
+from P4_bfs import bfs
+
+sc = SparkContext()
+textFile = sc.textFile('source.csv',20)
+
+#startNode = 'CAPTAIN AMERICA'		
+#startNode = 'MISS THING/MARY'
+startNode = 'ORWELL'
+graph = (
+	textFile
+	.map(lambda x:(x.split('"')[3],x.split('"')[1]))
+	.groupByKey()
+	.map(lambda t:(t[0],list(t[1])))
+	.flatMap(lambda f:[(y,tuple([x for x in f[1] if x != y])) for y in f[1]])
+	.groupByKey().map(lambda f:(f[0],10000,frozenset([inner for outer in f[1] for inner in outer]),False))
+	.map(lambda x:(x[0],0,x[2],True) if x[0] == startNode else x)
+	.cache()
```

Comment:
Looks like your code works, but it is extremely hard to follow. If your argument to a lambda function is a tuple, consider representing that as lambda (t1, t2, t3, ...) instead of lambda x and then having to access x[0] and x[1]. It makes code way more readable!